package/gluon-mesh-vpn-wireguard/files/lib/netifd/proto/gluon_wireguard.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
+# Copyright 2016-2017 Christof Schulze <christof@christofschulze.com>
+# Licensed to the public under the Apache License 2.0.
+
+. /lib/functions.sh
+. ../netifd-proto.sh
+init_proto "$@"
+
+proto_gluon_wireguard_init_config() {
+  no_device=1
+  available=1
+  renew_handler=1
+}
+
+proto_gluon_wireguard_renew() {
+  local config="$1"
+  echo wireguard RENEW: $@
+  # ifdump="$(ubus call network.interface dump)"
+}
+
+proto_gluon_wireguard_setup() {
+  local config="$1"
+  # ifname=$(ubus call network.interface dump | jsonfilter -e "@.interface[@.interface=\"$config\"]")
+  ifname=$(uci get network.$config.ifname) # we need uci here because nodevice=1 means the device is not part of the ubus structure
+  group=$(uci get wireguard.mesh_vpn.group)
+
+  if [ $(uci get wireguard.mesh_vpn_backbone.enabled) == 1 ]; then
+    ip link del $ifname
+    ip link add dev $ifname type wireguard
+    ip link set mtu $(uci get wireguard.mesh_vpn.mtu) dev $ifname
+    ip link set multicast on dev $ifname
```

Comment:
You will need something like pimd to have multicast connectivity between multiple interfaces.

Reply:
right. or continue to use mmfd.  set multicast may not be strictly needed right now after all.

Reply:
I think setting multicast here does not do anything as WireGuard doesn't support multicast packets. Ockham's razor applies here to reduce confusion. Please remove it, if it doesn't do anything.

Reply:
the flag is needed to make babel work